src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UsePropertyInsteadOfCountMethodWhenAvailableTests.cs

```diff
@@ -646,5 +647,75 @@ public class SomeClass
                     .WithLocation(8, 23)
                     .WithArguments(nameof(IReadOnlyCollection<int>.Count)));
         }
+
+        [Fact, WorkItem(3724, "https://github.com/dotnet/roslyn-analyzers/issues/3724")]
+        public static async Task PropertyAccessParentIsNotAlwaysDirectlyTheInvocation()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class C
+{
+    public static bool IsChildPath(string parentPath, string childPath, IEnumerable<string> obj)
+    {
+        return (IsDirectorySeparator(childPath[parentPath.Length]) ||
+            IsDirectorySeparator(childPath[{|CA1829:parentPath.Count()|}])) ||
+            0.Equals((uint)obj.Count());
```

Comment:
You should remove this here and in VB tests, it causes diagnosis of CA1827 that this test does not expect.

The reason for CA1827 being returned and not CA1829 is that `IEnumerable<T>` does not contain a `Count` or `Length` property.

Reply:
This is actually the test failing that I was talking about. `CA1827` suggests to use `.Any()` here but it shouldn't! I cannot use `Any` when I actually need the count value.

Reply:
I don't get why it shouldn't. As I said earlier, there is no `Count` property in `IEnumerable` nor `IEnumerable<T>`.

Reply:
Oh wait I am stupid it is actually valid!

Reply:
Given that `obj` is an `IEnumerable<T>`, I think there is no need to test for `0.Equals(obj.Count())` since that would report CA1827 and these tests focus primarily on rule CA1829.

Up to you if you want to keep it, just make sure to fix the CI issues.

Reply:
I have removed the test but I think we should create a separate ticket to fix the code/fixer but this test case because it's not handled properly.

Example:
```csharp
[Fact]
public Task ZeroEqualsUIntCastCount_WithPredicate_Diagnostic_CA1827()
	=> VerifyCodeFixAsync<UseCountProperlyAnalyzer, CSharpDoNotUseCountWhenAnyCanBeUsedFixer>(
		testSource: _sourceProvider.GetCodeWithExpression(
			_sourceProvider.GetEqualsUintCastTargetExpressionInvocationCode(0, withPredicate: true, _sourceProvider.MethodName),
			_sourceProvider.ExtensionsNamespace),
		CSharpCodeFixVerifier<UseCountProperlyAnalyzer, CSharpDoNotUseCountWhenAnyCanBeUsedFixer>
		.Diagnostic(UseCountProperlyAnalyzer.s_rule_CA1827)
		.WithArguments("Count")
		.WithSpan(10, 21, 10, 63),
		fixedSource: _sourceProvider.GetCodeWithExpression(
			_sourceProvider.GetFixedExpressionCode(withPredicate: true, negate: true),
			_sourceProvider.ExtensionsNamespace));
```
	
```csharp		
internal string GetEqualsUintCastTargetExpressionInvocationCode(int value, bool withPredicate, string methodName)
	=> $@"{value}.Equals({GetUIntCastCode(GetTargetExpressionCode(withPredicate, methodName))})";

internal abstract string GetUIntCastCode(string expression);
```

```csharp
internal override string GetUIntCastCode(string expression)
	=> $"(uint){expression}";
```

```csharp
internal override string GetUIntCastCode(string expression)
	=> $"CUInt({expression})";
```

Reply:
And btw, IMO, this test infra is a lot harder to use and reason about that a plain string test.

Reply:
> Could you please file the issue?

Issue #3743

> I found it useful to test the same scenarios for diferent types and for both VB and C# languages since it would avoid test duplication.

I agree that being able to reuse the code with various type is useful but the current infra makes it really hard (for me) to read and debug. We could use strings with interpolated strings where we would inject the type but keeping the overall test string easily readable. Again that's only my opinion and the goal is not to be mean or give a bad criticism.